### PR TITLE
[googlepay] add offer related fields to type definitions

### PR DESCRIPTION
**Pull request by socsieng**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developers.google.com/pay/api/web/reference/request-objects#OfferDetail
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/googlepay/index.d.ts [hunk 1]
```diff
@@ -94,6 +94,11 @@ declare namespace google.payments.api {
          */
         transactionInfo: TransactionInfo;
 
+        /**
+         * Offers available for redemption that can be used with current order.
```
Comment (JlUgia) [inline types/googlepay/index.d.ts:98]: with **the** current order?

File: types/googlepay/index.d.ts [hunk 2]
```diff
@@ -240,6 +245,12 @@ declare namespace google.payments.api {
          */
         paymentMethodData: PaymentMethodData;
 
+        /**
+         * Contains the data for offer applied by the user. This will be
```
Comment (JlUgia) [inline types/googlepay/index.d.ts:249]: Missing `the` here too?

File: types/googlepay/index.d.ts [hunk 3]
```diff
@@ -1146,6 +1172,45 @@ declare namespace google.payments.api {
...
+         */
+        offers: OfferDetail[];
+    }
+
+    /**
+     * Definition for each offer to be applied to this Payment Request.
+     */
+    interface OfferDetail {
+        /**
+         * Redemption code available for this transaction. This would be used to
```
Comment (JlUgia) [inline types/googlepay/index.d.ts:1191]: Maybe use more direct tenses to simplify the wording:
`This is used to identify the offer when the user decides...`

File: types/googlepay/index.d.ts [hunk 4]
```diff
@@ -1146,6 +1172,45 @@ declare namespace google.payments.api {
...
+         * identify the offer in case the user decides to apply the offer.
+         */
+        redemptionCode: string;
+
+        /**
+         * Description for the offer visible to the user to inform them about
+         * the offer. The description would be displayed in buyflow and should
```
Comment (JlUgia) [inline types/googlepay/index.d.ts:1198]: How about: The description `is` displayed...?

File: types/googlepay/index.d.ts [hunk 5]
```diff
@@ -1146,6 +1172,45 @@ declare namespace google.payments.api {
...
+         * be less than 60 characters long.
+         */
+        description: string;
+    }
+
+    /**
+     * Definition for each offer to be applied to this Payment Request.
```
Comment (JlUgia) [inline types/googlepay/index.d.ts:1205]: If it's not a particular "Payment Request", I'd suggest we "lowercase it".

**Comment by JlUgia** [comment]
Added a couple of wording suggestions.
Everything else LGTM

File: types/googlepay/index.d.ts (hunk 1 above)
Comment (socsieng) [reply types/googlepay/index.d.ts:98]: Will apply in a separate PR

File: types/googlepay/index.d.ts (hunk 2 above)
Comment (socsieng) [reply types/googlepay/index.d.ts:249]: Will apply in a separate PR

File: types/googlepay/index.d.ts (hunk 3 above)
Comment (socsieng) [reply types/googlepay/index.d.ts:1191]: Will apply in a separate PR

File: types/googlepay/index.d.ts (hunk 4 above)
Comment (socsieng) [reply types/googlepay/index.d.ts:1198]: Will apply in a separate PR

File: types/googlepay/index.d.ts (hunk 5 above)
Comment (socsieng) [reply types/googlepay/index.d.ts:1205]: Will apply in a separate PR

**Comment by JlUgia** [comment]
LGTM

**Review by JlUgia — APPROVED** [review]

**Comment by socsieng** [comment]
Ready to merge
